src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -145,15 +145,10 @@ namespace OpenMS
       const double rt_left_lim = spectrum_rt - rt_window_ / 2.0;
       const double rt_right_lim = spectrum_rt + rt_window_ / 2.0;
       const std::vector<Precursor> precursors = spectrum.getPrecursors();
-      if (precursors.size() < 1)
-      {
-        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
-                                         "Spectrum does not contain precursor info.");
-      }
-      const double spectrum_mz = precursors[0].getMZ();
+      const double spectrum_mz = precursors.empty() ? 0.0 : precursors.front().getMZ();
       const double mz_tolerance = mz_unit_is_Da_ ? mz_tolerance_ : mz_tolerance_ / 1e6;
-      const double mz_left_lim = spectrum_mz - mz_tolerance;
-      const double mz_right_lim = spectrum_mz + mz_tolerance;
+      const double mz_left_lim = spectrum_mz ? spectrum_mz - mz_tolerance : std::numeric_limits<double>::min();
```

Comment:
Basically the problem arised when I tried to use this class with MS1 data.

With MS1 there is no precursor information. That said, I would have no value to assign to `spectrum_mz`. I assigned a value of `0.0` in code.

What happens, though, is that the condition `target_mz >= mz_left_lim && target_mz <= mz_right_lim` would never pass (unless `target_mz` happens to be a value within the tolerance range).

So what I did is just disable the mz check in case of MS1 data (or, generally, when there is no precursor info).

Is this solution scientifically legit, or am I doing something undesired?

Another solution I could think of is to initialize `spectrum_mz` to the actual (not the precursor) mz. But both solutions would just be workarounds to that mz check.